pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/HttpSegmentFetcher.java

```diff
@@ -26,15 +26,38 @@
 public class HttpSegmentFetcher implements SegmentFetcher {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(HttpSegmentFetcher.class);
+  private static final String MAX_RETRIES = "maxRetries";
+  private static final int DEFAULT_MAX_RETRIES = 3;
+  private int maxRetries = DEFAULT_MAX_RETRIES;
 
   @Override
   public void init(Map<String, String> configs) {
+    if (configs.containsKey(MAX_RETRIES)) {
+      try {
+        maxRetries = Integer.parseInt(configs.get(MAX_RETRIES));
+      } catch (Exception e) {
+        maxRetries = DEFAULT_MAX_RETRIES;
+      }
+    }
   }
 
   @Override
   public void fetchSegmentToLocal(String uri, File tempFile) throws Exception {
-    final long httpGetResponseContentLength = FileUploadUtils.getFile(uri, tempFile);
-    LOGGER.info("Downloaded file from {} to {}; Length of httpGetResponseContent: {}; Length of downloaded file: {}", uri, tempFile,
-        httpGetResponseContentLength, tempFile.length());
+    for (int retry = 1; retry <= maxRetries; ++retry) {
```

Comment:
When we get a segment refresh message from the controller, we do not want to retry multiple times. That will hold up other segment refresh messages.